cloudfour/cloudfour.com-patterns
# Add subheading support to Heading

## Overview

This PR refactors our heading component class and template to accept a `subheading` option.

It differs slightly from the prototype in that the subheading has a `medium` font weight instead of `normal` because I thought that made the two segments of the heading appear more closely related.

The reasons for the refactor are as follows:

- The previous structure made it challenging to correctly align a permalink if a subheading is present.
- The previous template assumed that permalinks were the sole reason the complexity of the markup might change, but subheadings complicated this.
- The previous stories, which each had their own separate demo file, made testing various combinations of options challenging, leading me to rewrite the stories using controls.
- Because the `content` and `subheading` options may be properties or blocks, I found it necessary to add some additional `makeTwig*` utility functions for defining the `templateSource` intelligently.

I marked this as a `minor` change because the template syntax has not changed.

See the docs and code comments for specific implementation details.

## Screenshots

![Screenshot 2021-08-17 at 12-48-17 Storybook](https://user-images.githubusercontent.com/69633/129792153-73eac613-662e-4dfc-a5a4-718542b6bb00.png)

## Testing

[On the deploy preview...](https://deploy-preview-1489--cloudfour-patterns.netlify.app/)

1. Verify that Heading stories work as intended
2. Check that the Heading story code samples change accurately depending on controls
3. Verify that the Speaking Event prototype's "Watch" heading and subheading work as intended

---

- Fixes #1464 

/CC @dromo77 @calebeby
